test/e2e/safari/webview/web-specs.js

```diff
@@ -0,0 +1,29 @@
+import { IosDriver } from '../../../../lib/driver';
+import { util } from 'appium-support';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('Web', function () {
+  let driver;
+  beforeEach(function () {
+    driver = new IosDriver();
+  });
+  describe('.convertElementsForAtoms', function () {
+    it('should not alter primitives', function () {
+      let primitives = ['hello', 1, true];
+      driver.convertElementsForAtoms(primitives).should.eql(primitives);
+    });
+    it('should parse W3C and MJSONWP element ids', function () {
+      for (const elType of ['ELEMENT', util.W3C_WEB_ELEMENT_IDENTIFIER]) {
+        driver.webElementIds = [123];
+        const args = [{
+          [elType]: 5000
+        }];
+        driver.convertElementsForAtoms(args).should.eql([{ELEMENT: 123, [util.W3C_WEB_ELEMENT_IDENTIFIER]: 123}]);
```

Comment:
This line is breaking the build. I'm not sure what the intention is in `convertElementsForAtoms`, so I cannot fix it.

Reply:
I'll look at it today

Reply:
I'll look at it today

Reply:
the point of `convertElementForAtoms` is to unwrap an element, subtract 5000 from its numeric id, and rewrap it.

the reason is that the selenium atoms maintain their own element id list, which would conflict with the driver's element id cache. so we add 5000 to the selenium atom element id before returning it to the client. that means we also have to subtract it before reusing it in the atoms context.

Reply:
this test should probably be fixed by making the arg `5123` instead?

Reply:
I think the test should not have the W3C part in the expectation, since the atoms don't expect it. But that's the part where I don't know what the intention of the test or the changes to the function are.

Reply:
I put in a fix for that. The atom doesn't include the W3C element format, and I put in a fix to reflect that.